pkg/metrics/exporter.go

```diff
@@ -49,19 +54,48 @@ func RegisterPrometheusExporter(registry *prom.Registry) (http.Handler, error) {
 // RegisterStackdriverExporter register a Stackdriver exporter to OpenCensus.
 // It will add Agones metrics into Stackdriver on Google Cloud.
 func RegisterStackdriverExporter(projectID string) (sd *stackdriver.Exporter, err error) {
-	// Default project will be used
+	instanceID, err := metadata.InstanceID()
```

Comment:
Does this code only run on GCP?

If not, I wonder if we should allow a user to customize the instance / zone / cluster-name values via environment variables for aggregation purposes.

Reply:
No technically you can run this outside of GCP. However if you do you won't be able to correlate with existing GCP workload because `instance_id`,`zone` and `cluster_name` won't be valid which makes sense.

I think we should allow users to be able to set `DefaultMonitoringLabels *Labels` via configuration, so they can tag their custom workload, but it wasn't there already so that can be done in another PR.

Reply:
ok let's do this.